lib/services/elasticsearch.js

```diff
@@ -153,1425 +168,1355 @@ class ElasticSearch extends Service {
 
   /**
-   * Empty the content of a collection. Keep the existing mapping.
+   * Empties the content of a collection. Keep the existing mapping.
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   *
+   * @returns {Promise.<Object>} { }
    */
-  truncateCollection(request) {
-    const
-      deleteRequest = new Request({
-        index: request.input.resource.index,
-        collection: request.input.resource.collection,
-        body: {
-          query: {
-            match_all: {}
-          }
-        }
-      }, {
-        user: request.context.user
-      });
+  truncateCollection (index, collection) {
+    let mappings;
 
-    deleteRequest.input.args.refresh = request.input.args.refresh || false;
+    const esRequest = {
+      index: this._getESIndex(index, collection)
+    };
 
-    return this.deleteByQuery(deleteRequest)
-      .catch(error => Bluebird.reject(this.esWrapper.formatESError(error)));
+    return this.getMapping(index, collection)
+      .then(collectionMappings => {
+        mappings = collectionMappings;
+
+        return this.client.indices.delete(esRequest);
+      })
+      .then(() => this.client.indices.create({ ...esRequest, mappings }))
+      .then(() => {
+        // return nothing
+      })
+      .catch(error => this.esWrapper.reject(error));
   }
 
   /**
-   * Run several action and document
+   * Runs several action and document
    *
-   * @param {Request} request
-   * @returns {Promise}
+   * @param {String} index - Index name
+   * @param {String} collection - Collection name
+   * @param {Array<Object>} documents - Documents to import
+   * @param {Object} options - timeout (null), refresh (false), userId (null)
+   *
+   * @returns {Promise.<Object>} { items, errors }
    */
-  import(request) {
+  import (
+    index,
+    collection,
+    documents,
+    { refresh=false, timeout=null, userId=null } = {})
+  {
     const
+      esIndex = this._getESIndex(index, collection),
       actionNames = ['index', 'create', 'update', 'delete'],
-      esRequest = initESRequest(request,
-        ['consistency', 'refresh', 'timeout', 'fields']),
-      userId = this._getUserId(request),
-      dateNow = Date.now();
+      dateNow = Date.now(),
+      esRequest = {
+        body: documents,
+        refresh,
+        timeout
+      },
+      kuzzleMeta = {
+        created: {
+          author: getUserId(userId),
+          createdAt: dateNow,
+          updatedAt: null,
+          updater: null
+        },
+        updated: {
+          updater: getUserId(userId),
+          updatedAt: dateNow
+        }
+      };
 
     assertWellFormedRefresh(esRequest);
 
-    if (!request.input.body || !(request.input.body.bulkData instanceof Object)) {
-      return Bluebird.reject(
-        this.getError('missing_or_invalid_import_attribute')
-      );
-    }
+    let
+      actionCount = 0,
+      lastAction;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the declaration of "lastAction" to line 779. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2392)